nodejs/week3/routes.js

```diff
@@ -0,0 +1,29 @@
+const express = require("express");
+const contactsAPIRouter = express.Router();
+
+const knex = require("./database");
+
+contactsAPIRouter.get("/", async (req, res) => {
+    let query = knex.select("*").from("contacts");
+    
+    if ("sort" in req.query) {
+        const orderBy = req.query.sort.toString();
+        if (orderBy.length > 0) {
+        query = query.orderByRaw(orderBy);
+        }
+    }
```

Comment:
You have put in great effort in avoiding the sql injection vulnerability. And yes, I can see that you receive `sort column name` and `order` which has be handled. And I would like to suggest to refactor the code like the following, and also verify that the value of the `order` is `ASC` and `DESC`. and to deep dive, you can also verify the column name whether the column name matches the column name of the table or not, but not necessary, just for information.

`if('sort' in req.query){`
`const sort = req.query.sort.toString().trim();`
`const [column, order] = sort.split(' ');`
//  add if condition to verify whehter order matches value: ASC or DESC
`query=query.orderByRaw('?? ??' [column, order]); `  // knex parameterized query, check documentation 
`}`